src/optimizer/plan_generator.cpp

```diff
@@ -1063,4 +1065,90 @@ void PlanGenerator::Visit(const Analyze *analyze) {
                      .Build();
 }
 
+///////////////////////////////////////////////////////////////////////////////
+// CTE
+///////////////////////////////////////////////////////////////////////////////
+
+void PlanGenerator::Visit(const CteScan *cte_scan) {
+  // CteScan has the same output schema as the child plan!
+  NOISEPAGE_ASSERT(children_plans_.size() <= 2, "CteScan needs at most 2 child plans");
```

Comment:
similar to previous comment, when are 0/1/2 child plans OK?

Reply:
I added a comment in the code for this, but:
0 children: the node is not a leader (i.e. it reads from the temp table but does not write)
1 child: the node is the first time we refer to a _simple_ CTE table, in which case we populate the table as well (this is a leader); the child is the plan for the query used to populate the table
2 children: the node is a leader for an _inductive_ CTE table, in which case the two children are the plans for the base and inductive queries for the table